training/cleaning_pipeline.py

```diff
@@ -110,7 +110,11 @@ def filter(
     ) -> tuple[str | None, str | None, dict]:
         characters_to_rstrip = ["\n", "\t", "\r", ".", ","]
         subheading = subheading.strip()
-        description = " ".join(description.split())
+        description = (
+            " ".join(str(description).split())
```

Comment:
Although it technically works to put the input sanitisation here, it should really be done where it's actually being ingested, so that would be in the aws handler itself as that has the knowledge that the input is unsanitised.

This method is expecting a string and shouldn't really need to validate those types - it should be up to the first place where the unsanitised input comes in to do that type checking. The joys of dynamically typed languages, eh?!

That way, if we were to even remove this cleaning, then we wouldn't also inadvertently lose that input sanitisation which is being done here when it's not really this class's job.

Reply:
Putting this in the handler also means that you can (more correctly) send some kind of 400 error to the client if the description isn't a string.

Reply:
Ok yes, I see your point! I'll look into it!